server/src/document/DbtTextDocument.ts

```diff
@@ -177,7 +177,10 @@ export class DbtTextDocument {
 
     const jinjas = this.jinjaParser.findAllJinjaRanges(this.rawDocument);
 
-    return jinjas === undefined || (jinjas.length > 0 && this.jinjaParser.isJinjaModified(jinjas, changes));
+    return (
+      jinjas === undefined ||
+      (jinjas.length > 0 && (this.jinjaParser.isJinjaModified(jinjas, changes) || this.compiledDocument.getText() === this.rawDocument.getText()))
+    );
```

Comment:
If we can add some unit/e2e test for this case? (When document will be recompiled after any change).
`this.compiledDocument.getText() === this.rawDocument.getText()`

Reply:
Added unit test.